packages/context/src/binding.ts

```diff
@@ -215,17 +215,47 @@ export class Binding<T = BoundValue> {
     return this;
   }
 
-  tag(tagName: string | string[]): this {
-    if (typeof tagName === 'string') {
-      this.tags.add(tagName);
-    } else {
-      tagName.forEach(t => {
-        this.tags.add(t);
-      });
+  /**
+   * Tag the binding with names
+   * @param tagNames A list of names
+   *
+   * @example
+   * ```ts
+   * binding.tag('a');
+   * binding.tag('a', 'b');
+   * binding.tag(['a', 'b'], 'c')
```

Comment:
I find this syntax confusing. What does `['a', 'b']` mean? Is it creating a tag with name `a` and value `b`? Can we use an object with named properties instead please?

```ts
// add a single tag with no value (backwards compatiblity)
binding.tag('controller');
// add any number of tags with no value - do we want this?
binding.tag('controller', 'rest');
// add tags with values
binding.tag({controller: true, 'extensionPoint': 'controllers'});
```

Maybe your intention was different - could you please provide more details?

Reply:
The first round was to keep `tag` method only for names using rest params. It allows string or string[]. A new `tagValue` method was introduced to set name/value for tags.

With your feedback, I have consolidated both flavors into `tag` and allow tags to be set by name/value objects.